rust/lance-io/src/local.rs

```diff
@@ -111,13 +115,26 @@ impl Reader for LocalObjectReader {
     }
 
     /// Returns the file size.
-    async fn size(&self) -> Result<usize> {
-        Ok(self.file.metadata()?.len() as usize)
+    async fn size(&self) -> object_store::Result<usize> {
```

Comment:
thank you! for `object_store::Result`

Reply:
I still hate the error handling overall. Will refactor it later.